library/core/src/ptr/mod.rs

```diff
@@ -141,9 +134,7 @@ mod mut_ptr;
 ///
 /// Note that even if `T` has size `0`, the pointer must be non-NULL and properly aligned.
 ///
-/// [valid]: ../ptr/index.html#safety
-/// [`Copy`]: ../marker/trait.Copy.html
-/// [`write`]: ../ptr/fn.write.html
+/// [valid]: #safety
```

Comment:
```suggestion
/// [valid]: self#safety
```

and the same for the rest of these links.

Reply:
@jyn514 Would it be possible to add a warning if a link starts with `#`?

Reply:
Hmm, because it will be different on re-export. Yeah, that makes sense - it should recommend `Self` on a type and `self` on a module.

Reply:
By the way, why doesn't plain `#safety` work?

Reply:
Opened #76545.

Reply:
Actually thinking about it I'm not sure ... I thought on re-export it would break, but if the page is inlined it will still work (e.g. https://doc.rust-lang.org/std/ptr/fn.drop_in_place.html#safety).

@ollie27 why did you suggest adding `self`?

Reply:
@jyn514 I think that's a different header than the one that's linked to here though. That's `drop_in_place()#safety`, this is `self#safety`.

Reply:
Ohhh, that makes sense. And explains your question too: `#safety` links to the current page, which is drop_in_place.html, but `self#safety` links to the module.

Maybe https://github.com/rust-lang/rust/issues/76546 should be closed then, since it's not always applicable?

Reply:
Yeah, it make sense to close it.